embrace-android-sdk/src/main/java/io/embrace/android/embracesdk/internal/logs/EmbraceLogService.kt

```diff
@@ -0,0 +1,128 @@
+package io.embrace.android.embracesdk.internal.logs
+
+import io.embrace.android.embracesdk.LogExceptionType
+import io.embrace.android.embracesdk.Severity
+import io.embrace.android.embracesdk.capture.metadata.MetadataService
+import io.embrace.android.embracesdk.internal.utils.Uuid
+import io.embrace.android.embracesdk.session.id.SessionIdTracker
+import io.embrace.android.embracesdk.worker.BackgroundWorker
+import io.opentelemetry.api.common.AttributeKey
+import io.opentelemetry.api.logs.Logger
+import java.util.concurrent.TimeUnit
+
+/**
+ * Creates log records to be sent using the Open Telemetry Logs data model.
+ */
+internal class EmbraceLogService(
+    private val logger: Logger,
+    private val metadataService: MetadataService,
+    private val sessionIdTracker: SessionIdTracker,
+    private val backgroundWorker: BackgroundWorker
+) : LogService {
+
+    override fun log(
+        message: String,
+        severity: Severity,
+        properties: Map<String, Any>?
+    ) {
+        val sessionId = sessionIdTracker.getActiveSessionId()
+        val appState = metadataService.getAppState()
+
+        backgroundWorker.submit {
+            // TBD: Check if config enabled
+            // TBD: Check if log should be gated
+            // TBD: Count log and enforce limits
+
+            emitLog(
+                message,
+                severity,
+                sessionId,
+                appState,
+                properties
+            )
+        }
+    }
+
+    override fun logException(
+        message: String,
+        logExceptionType: LogExceptionType,
+        properties: Map<String, Any>?,
+        stackTraceElements: Array<StackTraceElement>?,
+        customStackTrace: String?,
+        context: String?,
+        library: String?,
+        exceptionName: String?,
+        exceptionMessage: String?
+    ) {
+        val sessionId = sessionIdTracker.getActiveSessionId()
+        val appState = metadataService.getAppState()
+
+        backgroundWorker.submit {
+            // TBD: Check if config enabled
+            // TBD: Check if log should be gated
+            // TBD: Count log and enforce limits
+
+            val attributes = mutableMapOf<String, Any>()
+            properties?.let { attributes.putAll(it) }
```

Comment:
It would be good to add a test for this defensive copy, if it's needed. Also, this is inconsistent with `log` which doesn't make a defensive copy

Reply:
Removed.